### PR TITLE
Fix for 2348 and 2344

### LINKED ISSUE #2348: Getters and setters made inaccessible
```javascript
Polymer({
	'is'		: 'cs-nav-tabs',
	'extends'	: 'nav',
	properties	: {
		active	: {
			observer	: 'active_changed',
			type		: Number
		}
	},
	ready : function () {
		if (!this.querySelector('button[active]')) {
			this.active = 0;
		}
	},
	active_changed : function () {
		var buttons = this.querySelectorAll('button');
		var button;
		for (var index = 0; index < buttons.length; ++index) {
			button = buttons[index];
			button.active = index == this.active;
		}
	}
});
```
```javascript
Polymer({
	'is'		: 'cs-button',
	'extends'	: 'button',
	properties	: {
		active	: {
			reflectToAttribute	: true,
			type				: Boolean
		}
	}
});
```
```html
<nav is="cs-nav-tabs">
	<button is="cs-button" type="button">One</button>
	<button is="cs-button" type="button">Two</button>
</nav>
```

At the moment when `button.active` is assigned (at `createdCallback` of `[is=cs-nav-tabs]`, you know), there are no getters/setters yet in `button` element, which eventually makes accessors below this newly declared property in prototypes chain and basically inaccessible.

**Firefox Nightly, Shadow DOM polyfill**

### PR DESCRIPTION
**Pull request by nazar-pc**
Fix for getters/setters for property become inaccessible when propery set on element before it is ready
Did not succeed in creating test case this time:(
Fixes #2348

### CODE REVIEW

**Comment by nazar-pc** [comment]
Indeed it fixes #2344 as well...

**Comment by tjsavage** [comment]
added @sjmiles to take a look on this as he was active on the [corresponding issue](https://github.com/Polymer/polymer/issues/2348)

File: src/standard/configure.html [hunk 1]
```diff
@@ -121,6 +121,9 @@
             value = value.call(this, this._config);
           }
           config[i] = value;
+        } else if (this.hasOwnProperty(i)) {
```
Comment (sorvell) [inline src/standard/configure.html]: `this[i] !== undefined` should be faster than `hasOwnProperty`.

**Comment by sorvell** [comment]
Sorry for the time lag on this. The main concern with this change is incurring a performance penalty on elements that do not have early-set properties. Would you be willing to do some performance testing on this? Thank.

**Comment by nazar-pc** [comment]
Modified as you suggested.
About performance: created 10 elements with 300 String properties there, 10 instances of each element on page (30k properties in total, all passing through newly added code).

Without patch `readystatechange` took 45.56ms in Chromium, with patch 79.45ms. This is 1.74 times slower, but since this step is executed only once for each element and takes so small amount of time even for 300 properties this is an affordable change.

Code I've used for testing: http://pastebin.com/ZdMnrZqN

File: src/standard/configure.html [hunk 2]
```diff
@@ -121,6 +121,9 @@
             value = value.call(this, this._config);
           }
           config[i] = value;
+        } else if (this[i] !== undefined) {
```
Comment (sorvell) [inline src/standard/configure.html]: Seems like the check `if` and the `else if` should be switched here. If there's an existing value, we should take that and not the default value.

**Comment by sorvell** [comment]
We plan to do some perf testing to see what overhead this adds to elements without early-property sets.

**Comment by nazar-pc** [comment]
Modified as suggested

**Comment by sorvell** [comment]
The performance of this change looks ok to me. Can you please add a test? Thanks.

**Comment by nazar-pc** [comment]
@sorvell, I've managed to write test for this change, also rebased against master

**Pull request merged**